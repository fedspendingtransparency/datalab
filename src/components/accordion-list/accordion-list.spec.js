import React from 'react';
import renderer from 'react-test-renderer';
import Accordion from './accordion-list';
import accordionStyles from './accordion-list.module.scss';

describe('Accordion', () => {
	const titleText = 'Placeholder Title';
	const bodyText = 'Placeholder Body';
	let component = renderer.create();
	renderer.act(() => {
		component = renderer.create(<Accordion title={titleText}>{bodyText}</Accordion>);
	});

	const instance = component.root;

	it('expect the snapshot to match', () => {
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('expects the title to be rendered.', () => {
		expect(instance.findByType('h1').children[0]).toBe(titleText);
	});

	it('expects the body content to be rendered', () => {
		expect(instance.findByProps({ className: accordionStyles.content }).children[0]).toBe(bodyText);
	});

	it('expects the accordion to toggle the "open" class when the button is clicked', async () => {
		const button = instance.findByType('button');
		const accordion = instance.findByType('section');
		expect(accordion.props.className).not.toContain(accordionStyles.open);

		renderer.act(() => {
			button.props.onClick({ stopPropagation: () => {} });
		});

		expect(accordion.props.className).toContain(accordionStyles.open);
	});
});