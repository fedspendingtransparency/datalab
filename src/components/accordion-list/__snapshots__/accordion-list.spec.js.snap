// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion expect the snapshot to match 1`] = `
<div
  className="container"
>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <section
        className="accordion"
        style={Object {}}
      >
        <h1
          className="heading"
          onClick={[Function]}
          style={Object {}}
        >
          Placeholder Title
          <button
            aria-label="Open toggle"
            className="toggle"
            onClick={[Function]}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-plus fa-w-14 plus"
              data-icon="plus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-minus fa-w-14 minus"
              data-icon="minus"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
        </h1>
        <div
          className="content"
        >
          Placeholder Body
        </div>
      </section>
    </div>
  </div>
</div>
`;
